client/src/components/ResourceSelector.tsx

```diff
@@ -0,0 +1,88 @@
+import { useEffect, useState } from "react";
+
+import { fetchWorkspaceResources } from "../services/api";
+import type { Workspace, WorkspaceResource } from "../types/api";
+
+interface ResourceSelectorProps {
+  workspace: Workspace;
+  onSelectResources: (resources: WorkspaceResource[]) => void;
+  selectedResources: WorkspaceResource[];
+}
+
+export function ResourceSelector({
+  workspace,
+  onSelectResources,
+  selectedResources,
+}: ResourceSelectorProps) {
+  const [resources, setResources] = useState<WorkspaceResource[]>([]);
+  const [loading, setLoading] = useState(true);
+  const [error, setError] = useState<string | null>(null);
+
+  useEffect(() => {
```

Comment:
I understand it might be too early for this yet, but have you considered moving such things into a standalone custom hook (fetch workspace resources)?

Reply:
Extracted this togic to hook